symengine/refine.cpp

```diff
@@ -0,0 +1,86 @@
+#include <symengine/refine.h>
+
+namespace SymEngine
+{
+
+void RefineVisitor::bvisit(const Abs &x)
+{
+    auto farg = x.get_arg();
+    auto newarg = apply(farg);
+    if (is_true(is_nonnegative(*newarg, assumptions_))) {
+        result_ = newarg;
+    } else if (is_true(is_nonpositive(*newarg, assumptions_))) {
+        result_ = neg(newarg);
+    } else if (is_a<Conjugate>(*newarg)) {
+        result_ = abs(down_cast<const Conjugate &>(*newarg).get_arg());
+    } else {
+        result_ = abs(newarg);
+    }
+}
+
+void RefineVisitor::bvisit(const Sign &x)
+{
+    auto farg = x.get_arg();
+    auto newarg = apply(farg);
+    if (is_true(is_positive(*newarg, assumptions_))) {
+        result_ = integer(1);
+    } else if (is_true(is_negative(*newarg, assumptions_))) {
+        result_ = integer(-1);
+        //    } else if (is_true(is_zero(*newarg, assumptions_))) {
```

Comment:
maybe remove these commented-out lines?

Reply:
Sorry about these. I didn't have the `is_zero` with assumptions when implementing this. I should uncomment and add some unit tests for this case.